azurerm/resource_arm_bot_channel_slack.go

```diff
@@ -0,0 +1,219 @@
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, botName, string(botservice.ChannelNameSlackChannel))
+	if err != nil {
+		return fmt.Errorf("Error making get request for Bot Channel Slack (Resource Group %q / Bot %q): %+v", resourceGroup, botName, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Bot Channel Slack (Resource Group %q / Bot %q): %+v", resourceGroup, botName, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmBotChannelSlackRead(d, meta)
+}
+
+func resourceArmBotChannelSlackRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).bot.ChannelClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	botName := id.Path["botServices"]
+	resp, err := client.Get(ctx, id.ResourceGroup, botName, string(botservice.ChannelNameSlackChannel))
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Error reading Bot Channel Slack (Resource Group %q / Bot %q): %+v", id.ResourceGroup, botName, err)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Bot Channel Slack (Resource Group %q / Bot %q): %+v", id.ResourceGroup, botName, err)
+	}
+
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", resp.Location)
+	d.Set("bot_name", botName)
+
+	if props := resp.Properties; props != nil {
+		if channel, ok := props.AsSlackChannel(); ok {
+			if channel != nil {
+				if channelProps := channel.Properties; channelProps != nil {
+					d.Set("client_id", channelProps.ClientID)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceArmBotChannelSlackUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).bot.ChannelClient
+	ctx := meta.(*ArmClient).StopContext
+
+	botName := d.Get("bot_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	channel := botservice.BotChannel{
+		Properties: botservice.SlackChannel{
+			Properties: &botservice.SlackChannelProperties{
+				ClientID:                utils.String(d.Get("client_id").(string)),
+				ClientSecret:            utils.String(d.Get("client_secret").(string)),
+				VerificationToken:       utils.String(d.Get("verification_token").(string)),
+				LandingPageURL:          utils.String(d.Get("landing_page_url").(string)),
+				IsEnabled:               utils.Bool(true),
+				RegisterBeforeOAuthFlow: utils.Bool(true),
+			},
+			ChannelName: botservice.ChannelNameSlackChannel1,
+		},
+		Location: utils.String(d.Get("location").(string)),
+		Kind:     botservice.KindBot,
+	}
+
+	if _, err := client.Update(ctx, resourceGroup, botName, botservice.ChannelNameSlackChannel, channel); err != nil {
+		return fmt.Errorf("Error issuing create request for Bot Channel Slack (Resource Group %q / Bot %q): %+v", resourceGroup, botName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, botName, string(botservice.ChannelNameSlackChannel))
```

Comment:
(same here) do we need to cast this to a string in all the calls? it's odd it's inconsistent

Reply:
Only because the SDK is inconsistent